### PR TITLE
fix(k8s): helm returned deprecated manifest version for tiller

### LINKED ISSUE #1166: with minikube 1.4.0.beta
### Current Behavior

`garden deploy`

```
[2019-09-08T05:29:17.561Z] Error: Unrecognized resource type extensions/v1beta1/Deployment
    at KubeApi.<anonymous> (/snapshot/dist/build/src/plugins/kubernetes/api.js:0)
    at Generator.next (<anonymous>)
    at fulfilled (/snapshot/dist/build/src/plugins/kubernetes/api.js:0)
Error Details:
manifest:
  apiVersion: extensions/v1beta1
  kind: Deployment
  metadata:
    creationTimestamp: null
    labels:
      app: helm
      name: tiller
    name: tiller-deploy
    namespace: demo-project
  spec:
    replicas: 1
    strategy: {}
    template:
      metadata:
        creationTimestamp: null
        labels:
          app: helm
          name: tiller
      spec:
        automountServiceAccountToken: true
        containers:
          - env:
              - name: TILLER_NAMESPACE
                value: demo-project
              - name: TILLER_HISTORY_MAX
                value: '0'
            image: 'gcr.io/kubernetes-helm/tiller:v2.14.1'
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /liveness
                port: 44135
              initialDelaySeconds: 1
              timeoutSeconds: 1
            name: tiller
            ports:
              - containerPort: 44134
                name: tiller
              - containerPort: 44135
                name: http
            readinessProbe:
              httpGet:
                path: /readiness
                port: 44135
              initialDelaySeconds: 1
              timeoutSeconds: 1
            resources: {}
        serviceAccountName: garden-tiller
  status: {}


[2019-09-08T05:29:17.579Z] Error: Failed resolving one or more provider configurations:
- local-kubernetes: Unrecognized resource type extensions/v1beta1/Deployment
    at Garden.<anonymous> (/snapshot/dist/build/src/garden.js:0)
    at Generator.next (<anonymous>)
    at fulfilled (/snapshot/dist/build/src/garden.js:0)
Error Details:
rawConfigs:
  - name: exec
  - name: container
  - name: local-kubernetes
    environments:
      - local
taskResults:
  resolve-provider.exec:
    type: resolve-provider
    key: resolve-provider.exec
    name: exec
    description: resolving provider exec
    output:
      name: exec
      dependencies: []
      moduleConfigs: []
      config:
        name: exec
        path: /Users/mike/tmp/garden/examples/demo-project
      status:
        ready: true
        outputs: {}
    dependencyResults: {}
  resolve-provider.container:
    type: resolve-provider
    key: resolve-provider.container
    name: container
    description: resolving provider container
    output:
      name: container
      dependencies: []
      moduleConfigs: []
      config:
        name: container
        path: /Users/mike/tmp/garden/examples/demo-project
      status:
        ready: true
        outputs: {}
    dependencyResults: {}
  resolve-provider.local-kubernetes:
    type: resolve-provider
    description: resolving provider local-kubernetes
    key: resolve-provider.local-kubernetes
    name: local-kubernetes
    error:
      detail:
        manifest:
          apiVersion: extensions/v1beta1
          kind: Deployment
          metadata:
            creationTimestamp: null
            labels:
              app: helm
              name: tiller
            name: tiller-deploy
            namespace: demo-project
          spec:
            replicas: 1
            strategy: {}
            template:
              metadata:
                creationTimestamp: null
                labels:
                  app: helm
                  name: tiller
              spec:
                automountServiceAccountToken: true
                containers:
                  - env:
                      - name: TILLER_NAMESPACE
                        value: demo-project
                      - name: TILLER_HISTORY_MAX
                        value: '0'
                    image: 'gcr.io/kubernetes-helm/tiller:v2.14.1'
                    imagePullPolicy: IfNotPresent
                    livenessProbe:
                      httpGet:
                        path: /liveness
                        port: 44135
                      initialDelaySeconds: 1
                      timeoutSeconds: 1
                    name: tiller
                    ports:
                      - containerPort: 44134
                        name: tiller
                      - containerPort: 44135
                        name: http
                    readinessProbe:
                      httpGet:
                        path: /readiness
                        port: 44135
                      initialDelaySeconds: 1
                      timeoutSeconds: 1
                    resources: {}
                serviceAccountName: garden-tiller
          status: {}
      type: kubernetes
messages:
  - '- local-kubernetes: Unrecognized resource type extensions/v1beta1/Deployment'

```

### Suggested solution(s)

Update `Deployment` to `v1`

### Additional context

using minikube 1.4.0 (with k8s 1.16) get an error. Looks like k8s 1.16 deprecated Deployment/v1beta1

with minikube 1.3.1 ( with k8s 1.15.2) everything works.

### Your environment

`garden version`
0.10.8

`kubectl version`
1.15.2
`docker version`
Docker version 19.03.2, build 6a30dfc

### PR DESCRIPTION
**Pull request by edvald**
**What this PR does / why we need it**:

Helm was returning an out-of-date manifest version for Tiller, which caused issues with Kubernetes 1.16. We patch it by converting the manifest version ourselves.

**Which issue(s) this PR fixes**:

Fixes #1166

### CODE REVIEW

File: garden-service/src/plugins/kubernetes/util.ts [hunk 1]
```diff
@@ -288,3 +288,38 @@ export function prepareEnvVars(env: ContainerEnvVars): V1EnvVar[] {
...
+      case "PodSecurityPolicy":
+        manifest.apiVersion = "policy/v1beta1"
+        break
+    }
+  }
+
+  // apps/v1/Deployment requires spec.selector to be set
+  if (kind === "Deployment") {
+    if (manifest.spec && !manifest.spec.selector) {
+      manifest.spec.selector = {
```
Comment (eysi09) [inline garden-service/src/plugins/kubernetes/util.ts]: Why the spread? Isn't this just the same as:
```
manifest.spec.selector = { matchLabels: manifest.spec.template.metadata.labels || manifest.metadata.labels }
```

In which case I'd suggest skipping the spread operator.

Comment (edvald) [reply garden-service/src/plugins/kubernetes/util.ts]: It's not the same, it's a (possibly too hard to read) shorthand that omits the property if the value is undefined. I can make it more verbose and straightforward if you like though.

Comment (eysi09) [reply garden-service/src/plugins/kubernetes/util.ts]: A comment would also be good. IMO it's fine to utilise the expressiveness of the language and just throw in a comment for those not familiar.

Comment (edvald) [reply garden-service/src/plugins/kubernetes/util.ts]: @eysi09 updated with a comment.

**Pull request merged**